File: Python/Software Development Module - Python/Week9/Average.py
"""
Average.py
Billy Cussen
01/10/2020
"""

class myAverage:
    def __init__(self, arrayNumbers):
        self.arrayNumbers = arrayNumbers
        self.average = 0
        self.min = 0
        self.max = 0
    
    def calculateAverage(self):
        total = 0
        self.min = self.arrayNumbers[0]
        self.max = self.arrayNumbers[0]
        for i in self.arrayNumbers:
            total+=i
            if i < self.min:
                self.min = i
            if i > self.max:
                self.max = i
        
        self.average = total / len(self.arrayNumbers)
        
        

File: Java/Software Development Module - Java/Week9/Average.java
/*
Average.java
Billy Cussen
18/07/2020
*/

public class Average {
    
    //Data Members
    private int[] input;
    private int average;
    private int min;
    private int max;
    private String output;

    //Constructor & Setter
    public Average(int[] input){
        this.input = input;
        this.average = average;
        this.min = 0;
        this.max = 0;
        this.output = new String();
    }

    //Compute & Getter
    public String computeAverage(){

        int total = 0;
        this.min = this.input[0];
        this.max = this.input[0];

        for(int i = 0; i < this.input.length;i++){

            total += this.input[i];

            if(this.input[i] < this.min){
                this.min = this.input[i];
            }

            if(this.input[i] > this.max){
                this.max = this.input[i];
            }

        }

        this.average = total/this.input.length;
        this.output = "Average of numbers submitted: "+this.average+"\nSmallest Number included: "+this.min+"\nLargest Number included: "+this.max;
        return this.output;
    } 

}